src/main/java/picard/vcf/GenotypeConcordance.java

```diff
@@ -301,6 +341,89 @@ private boolean indexExists(final File vcf) {
+                if (tuple.leftVariantContext.isPresent()) {
+                    truthContext = tuple.leftVariantContext.get();
+                }
+                if (tuple.rightVariantContext.isPresent()) {
+                    callContext = tuple.rightVariantContext.get();
+                }
+
+                // Get the alleles for each genotype
+                final Alleles alleles = normalizeAlleles(truthContext, TRUTH_SAMPLE, callContext, CALL_SAMPLE);
+
+                // Ignore if both are one of mixed, filtered, or missing.
+                if (!alleles.allAlleles.isEmpty()) {
+                    final List<Allele> allAlleles   = alleles.asAlleles();
+                    final List<Allele> truthAlleles = alleles.truthAlleles();
+                    final List<Allele> callAlleles  = alleles.callAlleles();
+
+                    // Initialize the variant context builder
+                    if (truthContext != null) {
+                        builder = new VariantContextBuilder(truthContext.getSource(), truthContext.getContig(), truthContext.getStart(), truthContext.getEnd(), Collections.emptyList());
+                        builder.computeEndFromAlleles(allAlleles, truthContext.getStart());
+                    }
+                    else if (callContext != null) {
+                        builder = new VariantContextBuilder(callContext.getSource(), callContext.getContig(), callContext.getStart(), callContext.getEnd(), Collections.emptyList());
+                        builder.computeEndFromAlleles(allAlleles, callContext.getStart());
+                    }
+                    else {
+                        throw new IllegalStateException("Both truth and call contexts are null!");
+                    }
+
+                    // Add the genotypes
+                    if (truthContext != null && !truthAlleles.isEmpty()) {
+                        final Genotype truthGenotype = truthContext.getGenotype(TRUTH_SAMPLE);
+                        final GenotypeBuilder genotypeBuilder = new GenotypeBuilder(truthGenotype);
+                        genotypeBuilder.name(outputTruthSampleName);
+                        genotypeBuilder.alleles(truthAlleles);
+                        if (!truthGenotype.hasAnyAttribute(VCFConstants.GENOTYPE_KEY)) {
+                            genotypeBuilder.attribute(VCFConstants.GENOTYPE_KEY, MISSING_VALUE_v4);
+                        }
+                        genotypes.add(genotypeBuilder.make());
+                    }
+                    else {
+                        final GenotypeBuilder genotypeBuilder = new GenotypeBuilder(outputTruthSampleName);
+                        genotypeBuilder.alleles(Arrays.asList(Allele.NO_CALL, Allele.NO_CALL));
+                        genotypeBuilder.attribute(VCFConstants.GENOTYPE_KEY, MISSING_VALUE_v4);
+                        genotypes.add(genotypeBuilder.make());
+                    }
+                    if (callContext != null && !callAlleles.isEmpty()) {
```

Comment:
This whole `if` block is repetitious.  Could be extracted to a method that takes in the sample name, genotype and set of alleles...